charms/kserve-controller/tests/unit/test_charm.py

```diff
@@ -23,12 +23,33 @@
     "services": {
         "kserve-controller": {
             "command": "/manager --metrics-addr=:8080",
-            "environment": {"POD_NAMESPACE": None, "SECRET_NAME": "kserve-webhook-server-cert"},
+            "environment": {
+                "POD_NAMESPACE": None,
+                "SECRET_NAME": "kserve-webhook-server-cert",
+            },
             "override": "replace",
             "startup": "enabled",
             "summary": "KServe Controller",
         }
-    }
+    },
+    "checks": {
+        "kserve-controller-ready": {
+            "override": "replace",
+            "level": "ready",
+            "period": "5m",
+            "timeout": "60s",
+            "threshold": 3,
+            "http": {"url": "http://localhost:8081/readyz"},
```

Comment:
suggestion: link to where the port and endpoint are defined upstream? just so we know the source of those

Reply:
Done!

9e73dc8